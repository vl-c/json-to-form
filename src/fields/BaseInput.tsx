import * as React from 'react';

export interface BaseInputProps {
  id: string;
  type: string;
  label?: string;
  value?: any;
  required?: boolean;
  disabled?: boolean;
  readonly?: boolean;
}

const BaseInput = (props: BaseInputProps): JSX.Element => {
  const { id, type, label, value, readonly, disabled } = props;

  return (
    <div className="input-block">
      {label && (
        <label className="input-label" htmlFor={id}>
          {label}
        </label>
      )}
      <input
        className="input-field"
        type={type}
        id={id}
        name={id}
        readOnly={readonly}
        disabled={disabled}
        defaultValue={value}
      />
    </div>
  );
};

export default BaseInput;
