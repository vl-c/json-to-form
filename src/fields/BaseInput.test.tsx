import * as React from 'react';

import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import BaseInput from './BaseInput';

function renderText() {
  return shallow(
    <BaseInput
      id={'base'}
      type={'text'}
      label={'BaseInput text'}
      value={'default value'}
    />
  );
}

function renderNumber() {
  return shallow(
    <BaseInput
      id={'base'}
      type={'number'}
      label={'BaseInput number'}
      value={1000}
    />
  );
}

it('BaseInput text shallow render', () => {
  const res = renderText();
  expect(shallowToJson(res)).toMatchSnapshot();
});

it('BaseInput number shallow render', () => {
  const res = renderNumber();
  expect(shallowToJson(res)).toMatchSnapshot();
});
