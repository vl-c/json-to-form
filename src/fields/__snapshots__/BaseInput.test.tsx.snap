// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseInput number shallow render 1`] = `
<div
  className="input-block"
>
  <label
    className="input-label"
    htmlFor="base"
  >
    BaseInput number
  </label>
  <input
    className="input-field"
    defaultValue={1000}
    id="base"
    name="base"
    type="number"
  />
</div>
`;

exports[`BaseInput text shallow render 1`] = `
<div
  className="input-block"
>
  <label
    className="input-label"
    htmlFor="base"
  >
    BaseInput text
  </label>
  <input
    className="input-field"
    defaultValue="default value"
    id="base"
    name="base"
    type="text"
  />
</div>
`;
